src/zarr/core/array.py

```diff
@@ -113,7 +114,7 @@ async def create(
         store: StoreLike,
         *,
         # v2 and v3
-        shape: ChunkCoords,
+        shape: int | ChunkCoords,
```

Comment:
Over in https://github.com/zarr-developers/zarr-python/pull/2128 @jhamman is adding a type called `ShapeLike`, which is `int | tuple[int, ...]`. I'm not a huge fan of `ChunkCoords` (because "coordinates" is a rather loaded term, and chunks are not the only thing indexed by tuples of ints), so maybe before or after his PR lands we can migrate this type over to `ShapeLike`